method/link.js

```diff
@@ -15,25 +75,19 @@ module.exports = function Link (ssb) {
     if (child && !isGroup(child)) return cb(new Error(`findLinks expected a groupId for child, got ${child} instead.`))
     if (!parent && !child) return cb(new Error('findLinks expects a parent or child id to be given'))
 
-    const query = [{
-      $filter: {
-        value: {
-          content: {
-            type,
-            ...opts,
-            tangles: {
-              link: { root: null, previous: null }
-            }
-          }
-        }
-      }
-    }]
-
     pull(
-      // NOTE: using ssb-query instead of ssb-backlinks
-      // because the backlinks query will get EVERY message which contains the groupId in it, which will be a LOT for a group
-      // then filters that massive amount down to the ones which have the dest in the right place
-      ssb.query.read({ query }),
+      ssb.db.query(
+        where(
+          and(
+            dbType(type),
+            parent && equal(seekParent, parent, { indexType: 'parent', prefix: true }),
+            child && equal(seekChild, child, { indexType: 'child', prefix: true }),
+            equal(seekTanglesLinkRoot, null, { indexType: 'tanglesLinkRoot', prefix: true }),
+            equal(seekTanglesLinkPrevious, null, { indexType: 'tanglesLinkPrevious', prefix: true })
```

Comment:
:hot_pepper: 

I think this index might be redundant, as it's very unlikely to have `root: null` and `previous: [a, b]`
I think we could do a `pull.filter` check instead of building an index for all the previous fields....? butt I'm not an index expert?